app/js/main.js

```diff
@@ -22,16 +22,30 @@ const DRAG_OVER_CSS = 'drag-over';
 const DRAG_OVER_EVENT = 'dragover';
 const DRAG_START_EVENT = 'dragstart';
 
+const TOUCH_START_EVENT = 'touchstart';
+const TOUCH_MOVE_EVENT = 'touchmove';
+const TOUCH_END_EVENT = 'touchend';
+
 const fileDrop = document.getElementById('files');
 const imagesList = document.getElementById('images-list');
 const clearButton = document.getElementById('clear-button');
 const stitchButton = document.getElementById('stitch-button');
 const result = document.getElementById('result');
 const keepAspectCheckbox = document.getElementById('keep-aspect');
+const zoomSlider = document.getElementById('zoom-slider');
+const zoomValue = document.getElementById('zoom-value');
 
 let dragState = false;
 let dragSource = null;
 
+zoomSlider.addEventListener('input', () => {
+    const zoomLevel = zoomSlider.value;
+    zoomValue.textContent = `${zoomLevel}%`;
+    const canvas = result.querySelector(CANVAS_TAG);
```

Comment:
`canvas` should be declared outside. No need to query for each input event.

Reply:
We should put this to an issue.